pipelines/ceus/ceus.py

```diff
@@ -147,119 +147,76 @@ def generate_result_plots(self):
                 max_total = buildingtype_df[['Heating', 'Cooling'] + base_enduses].sum(axis=1).max()
                 max_val = 1 if max_total <= 1 else int(max_total) + 1
-                title = f'{str(city)}-{str(buildingtype)}'
+                title = f'CEUS-{str(city)}-{str(buildingtype)}'
                 buildingtype_df['Baseload'] = buildingtype_df[base_enduses].sum(axis=1)
                 buildingtype_df = buildingtype_df.iloc[:24]
                 buildingtype_df = buildingtype_df.reset_index()
                 plot = buildingtype_df[['Baseload', 'Heating', 'Cooling']].plot(title=title, grid=True, xticks=ticks, ylim=(0, max_val), linewidth=2, color=['black','red','blue'])
                 plt.xlabel('Hour-of-Day')
-                plt.ylabel('Load (pu. base total peak)')   
+                plt.ylabel('Load (pu. base total peak)')
                 fig = plot.get_figure()
-                image_index_based_name = '{0:0=4d}'.format(image_index)
-                fig.savefig(f'{loadshapes_plots_dir}/{image_index_based_name}.png')
+                fig.savefig(f'{loadshapes_plots_dir}/{title}.png')
                 plt.close(fig)
-                image_index += 1    
 
         logger.info('GENERATING CEUS COMPONENT PLOTS')
 
-        image_index = 0
         for idx, city in enumerate(components.target.unique()):
             city_df = components.loc[components.target == city]
             for zdx, buildingtype in enumerate(loadshapes.buildingtype.unique()):
                 buildingtype_df = city_df.loc[city_df.buildingtype == buildingtype]
                 max_total = buildingtype_df[plotting_components].sum(axis=1).max()
                 max_val = 1 if max_total <= 1 else int(max_total) + 1
                 for ydx, daytype in enumerate(buildingtype_df.daytype.unique()):
-                    title = f'{str(city)}-{buildingtype}-{str(daytype)}'
+                    title = f'CEUS-{str(city).split(",")[0]}_{str(city).split(",")[1]}-{str(buildingtype)}-{str(daytype)}'
                     day_df = buildingtype_df.loc[buildingtype_df.daytype == daytype]
                     day_df = day_df.append(day_df.iloc[0])
                     day_df = day_df.reset_index()
                     plot = day_df[plotting_components].plot(kind='area', title=title, grid=True, xticks=ticks, ylim=(0, max_val), linewidth=2, color=['green','yellow','brown','blue','grey','black','red'])
                     plt.xlabel('Hour-of-Day')
-                    plt.ylabel('Load (pu. summer total peak)')   
+                    plt.ylabel('Load (pu. summer total peak)')
+                    fig = plot.get_figure()
+                    fig.savefig(f'{components_plots_dir}/{title}.png')
+                    plt.close(fig)
+
+    def loadshapes_plotting(self, loadshapes, directory, base_enduses):
+        ######## Plotting helper function
+        import numpy as np
+        import matplotlib.pyplot as plt
```

Comment:
Don't import these at the function level, put these imports up top

Reply:
fixed